APIReviewFoxy.md

```diff
@@ -0,0 +1,601 @@
+
+**Suggested Action**: Create new Covariance representation (3x3 upper triangle and 6x6 upper triangle) as well as potential helper functions. Identify areas to leverage it and then determine a migration path. New messages can leverage it, but existing ones will have a long migration.
+
+Msg
+
+
+
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Accel.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Accel.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/AccelStamped.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/AccelStamped.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/AccelWithCovariance.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/AccelWithCovariance.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/AccelWithCovarianceStamped.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/AccelWithCovarianceStamped.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Inertia.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Inertia.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/InertiaStamped.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/InertiaStamped.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Point.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Point.msg)
+    *   Comment: Point and Vector3 are the same message
+    *   **Suggested Action**: None, these are the same data structure but semantically different.
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Point32.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Point32.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/PointStamped.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/PointStamped.msg)
+*   [https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Polygon.msg](https://github.com/ros2/common_interfaces/blob/master/geometry_msgs/msg/Polygon.msg)
+    *   Comment: Polygon uses Point32 instead of Point. Would expect two message types for consistency: Polygon and Polygon32
+    *   **Suggested Action**: None. For “completeness” we could fill out the data types. But in general we don’t want to have a “complete” set of messages we would rather have everyone using the same datatype. There are very few use cases where a polygon is large enough require 64 bit precision and this saves 50% on bandwidth. It’s the same compromise as is made for point clouds. 
```

Comment:
Just a note, a polygon does not need to be large to justify 64 bit. It can also have a reference frame far away (UTM).
Of course this can be mitigated using in-between TF frames, but that defeats the purpose a little.

Reply:
If you're going to be working on those scales I'd recommend making another datatype with semantic meaning to support those use cases. A float is able to maintain millimeter precision to over 8 km. This datatype is not complex to reproduce.